src/Services/QueuedJobService.php

```
@@ -1533,6 +1579,271 @@ private function addJobHandlersToLogger(LoggerInterface $logger, QueuedJob $job,
         }
     }
 
+    /**
+     * Find any eligible jobs that qualify for an automated job retry and schedule their retry
+     *
+     * @throws ValidationException
+     */
+    protected function retryEligibleJobs(int $queueType): void
+    {
+        // Find any broken jobs that qualify for an automated job retry
+        $jobRetrySentinel = (int) static::config()->get('job_retry_sentinel');
+        $jobRetryLimit = (int) static::config()->get('job_retry_limit');
+        $jobRetryStatus = (array) static::config()->get('job_retry_status');
+
+        $jobStatusMap = [];
+        $jobRetrySentinel = max(0, $jobRetrySentinel);
+
+        // Job retries feature is disabled
+        if ($jobRetryLimit <= 0) {
+            return;
+        }
+
+        $validJobStatuses = QueuedJobDescriptor::singleton()->getJobStatusValues();
+
+        foreach ($jobRetryStatus as $retryStatus => $targetStatus) {
+            // Skip any disabled status conditions
+            if (!$targetStatus) {
+                continue;
+            }
+
+            // Skip any invalid status conditions
+            if (!in_array($targetStatus, $validJobStatuses)) {
+                continue;
+            }
```

A warning should be logged (if not an exception thrown outright) and possibly also added to the job messages so that developers can be made aware that their configuration is invalid.

Silently failing makes it harder to debug (if devs even realise there's a problem to debug).